nextflow_schema.json

```diff
@@ -1,27 +1,32 @@
 of the FASTQs, and the libraries will\n    still be processed correctly.\n  - This also applies to the SeqType column, i.e. with the example above, if one\n    run is PE and one run is SE, you need to give fake lane IDs to one of the\n    runs as well.\n- All _BAM_ files must be specified as `SE` under `SeqType`.\n  - You should provide a small decoy reference genome with pre-made indices, e.g.\n    the human mtDNA or phiX genome, for the mandatory parameter `--fasta` in\n    order to avoid long computational time for generating the index files of the\n    reference genome, even if you do not actually need a reference genome for any\n    downstream analyses.\n- nf-core/eager will only merge multiple _lanes_ of sequencing runs with the\n  same single-end or paired-end configuration\n- Accordingly nf-core/eager will not merge _lanes_ of FASTQs with BAM files\n  (unless you use `--run_convertbam`), as only FASTQ files are lane-merged\n  together.\n- Same libraries that are sequenced on different sequencing configurations (i.e\n  single- and paired-end data), will be merged after mapping and will _always_\n  be considered 'paired-end' during downstream processes\n  - **Important** running DeDup in this context is _not_ recommended, as PE and\n    SE data at the same position will _not_ be evaluated as duplicates.\n    Therefore not all duplicates will be removed.\n  - When you wish to run PE/SE data together `-dedupper markduplicates` is\n    therefore preferred.\n  - An error will be thrown if you try to merge both PE and SE and also supply\n    `--skip_merging`.\n  - If you truly want to mix SE data and PE data but using mate-pair info for PE\n    mapping, please run FASTQ preprocessing mapping manually and supply BAM\n    files for downstream processing by nf-core/eager\n  - If you _regularly_ want to run the situation above, please leave a feature\n    request on github.\n- DamageProfiler, NuclearContamination, MTtoNucRatio and PreSeq are performed on\n  each unique library separately after deduplication (but prior same-treated\n  library merging).\n- nf-core/eager functionality such as `--run_trim_bam` will be applied to only\n  non-UDG (UDG_Treatment: none) or half-UDG (UDG_Treatment: half) libraries.\n- Qualimap is run on each sample, after merging of libraries (i.e. your values\n  will reflect the values of all libraries combined - after being damage trimmed\n  etc.).\n- Genotyping will be typically performed on each `sample` independently, as\n  normally all libraries will have been merged together. However, if you have a\n  mixture of single-stranded and double-stranded libraries, you will normally\n  need to genotype separately. In this case you **must** give each the SS and DS\n  libraries _distinct_ `Sample_IDs`; otherwise you will receive a `file\n  collision` error in steps such as `sexdeterrmine`, and then you will need to\n  merge these yourself. We will consider changing this behaviour in the future\n  if there is enough interest."
```

Comment:
In the schema this file doesn't load. Just need to use an absolute file path I think: `https://github.com/nf-core/eager/blob/master/docs/images/usage/merging_files.png`

```suggestion
                    "help_text": "There are two possible ways of supplying input sequencing data to nf-core/eager.\nThe most efficient but more simplistic is supplying direct paths (with\nwildcards) to your FASTQ or BAM files, with each file or pair being considered a\nsingle library and each one run independently. TSV input requires creation of an\nextra file by the user and extra metadata, but allows more powerful lane and\nlibrary merging.\n\n##### Direct Input Method\n\nThis method is where you specify with `--input`, the path locations of FASTQ\n(optionally gzipped) or BAM file(s). This option is mutually exclusive to the\n[TSV input method](#tsv-input-method), which is used for more complex input\nconfigurations such as lane and library merging.\n\nWhen using the direct method of `--input` you can specify one or multiple\nsamples in one or more directories files. File names **must be unique**, even if\nin different directories.  \n\nBy default, the pipeline _assumes_ you have paired-end data. If you want to run\nsingle-end data you must specify [`--single_end`]('#single_end')\n\nFor example, for a single set of FASTQs, or multiple paired-end FASTQ\nfiles in one directory, you can specify:\n\n```bash\n--input 'path/to/data/sample_*_{1,2}.fastq.gz'\n```\n\nIf you have multiple files in different directories, you can use additional\nwildcards (`*`) e.g.:\n\n```bash\n--input 'path/to/data/*/sample_*_{1,2}.fastq.gz'\n```\n\n> :warning: It is not possible to run a mixture of single-end and paired-end\n> files in one run with the paths `--input` method! Please see the [TSV input\n> method](#tsv-input-method) for possibilities.\n\n**Please note** the following requirements:\n\n1. Valid file extensions: `.fastq.gz`, `.fastq`, `.fq.gz`, `.fq`, `.bam`.\n2. The path **must** be enclosed in quotes\n3. The path must have at least one `*` wildcard character\n4. When using the pipeline with **paired end data**, the path must use `{1,2}`\n   notation to specify read pairs.\n5. Files names must be unique, having files with the same name, but in different\n   directories is _not_ sufficient\n   - This can happen when a library has been sequenced across two sequencers on\n     the same lane. Either rename the file, try a symlink with a unique name, or\n     merge the two FASTQ files prior input.\n6. Due to limitations of downstream tools (e.g. FastQC), sample IDs may be\n   truncated after the first `.` in the name, Ensure file names are unique prior\n   to this!\n7. For input BAM files you should provide a small decoy reference genome with\n   pre-made indices, e.g. the human mtDNA or phiX genome, for the mandatory\n   parameter `--fasta` in order to avoid long computational time for generating\n   the index files of the reference genome, even if you do not actually need a\n   reference genome for any downstream analyses.\n\n##### TSV Input Method\n\nAlternatively to the [direct input method](#direct-input-method), you can supply\nto `--input` a path to a TSV file that contains paths to FASTQ/BAM files and\nadditional metadata. This allows for more complex procedures such as merging of\nsequencing data across lanes, sequencing runs, sequencing configuration types,\nand samples.\n\n<p align=\"center\">\n  <img src=\"https://github.com/nf-core/eager/blob/master/docs/images/usage/merging_files.png\" alt=\"Schematic diagram indicating merging points of different types of libraries, given a TSV input. Dashed boxes are optional library-specific processes\" width=\"70%\">\n</p>\n\nThe use of the TSV `--input` method is recommended when performing\nmore complex procedures such as lane or library merging. You do not need to\nspecify `--single_end`, `--bam`, `--colour_chemistry`, `-udg_type` etc. when\nusing TSV input - this is defined within the TSV file itself. You can only\nsupply a single TSV per run (i.e. `--input '*.tsv'` will not work).\n\nThis TSV should look like the following:\n\n| Sample_Name | Library_ID | Lane | Colour_Chemistry | SeqType | Organism | Strandedness | UDG_Treatment | R1 | R2 | BAM |\n|-------------|------------|------|------------------|--------|----------|--------------|---------------|----|----|-----|\n| JK2782      | JK2782     | 1    | 4                | PE      | Mammoth  | double       | full          | [https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2782_TGGCCGATCAACGA_L008_R1_001.fastq.gz.tengrand.fq.gz](https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2782_TGGCCGATCAACGA_L008_R1_001.fastq.gz.tengrand.fq.gz) | [https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2782_TGGCCGATCAACGA_L008_R2_001.fastq.gz.tengrand.fq.gz](https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2782_TGGCCGATCAACGA_L008_R2_001.fastq.gz.tengrand.fq.gz) | NA  |\n| JK2802      | JK2802     | 2    | 2                | SE      | Mammoth  | double       | full          | [https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2802_AGAATAACCTACCA_L008_R1_001.fastq.gz.tengrand.fq.gz](https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2802_AGAATAACCTACCA_L008_R1_001.fastq.gz.tengrand.fq.gz) | [https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2802_AGAATAACCTACCA_L008_R2_001.fastq.gz.tengrand.fq.gz](https://github.com/nf-core/test-datasets/raw/eager/testdata/Mammoth/fastq/JK2802_AGAATAACCTACCA_L008_R2_001.fastq.gz.tengrand.fq.gz) | NA  |\n\nA template can be taken from\n[here](https://raw.githubusercontent.com/nf-core/test-datasets/eager/reference/TSV_template.tsv).\n\n> :warning: Cells **must not** contain spaces before or after strings, as this\n> will make the TSV unreadable by nextflow. Strings containing spaces should be\n> wrapped in quotes.\n\nWhen using TSV_input, nf-core/eager will merge FASTQ files of libraries with the\nsame `Library_ID` but different `Lanes` values after adapter clipping (and\nmerging), assuming all other metadata columns are the same. If you have the same\n`Library_ID` but with different `SeqType`, this will be merged directly after\nmapping prior BAM filtering. Finally, it will also merge BAM files with the same\n`Sample_ID` but different `Library_ID` after duplicate removal, but prior to\ngenotyping. Please see caveats to this below.\n\nColumn descriptions are as follows:\n\n- **Sample_Name:** A text string containing the name of a given sample of which\n  there can be multiple libraries. All libraries with the same sample name and\n  same SeqType will be merged after deduplication.\n- **Library_ID:** A text string containing a given library, which there can be\n  multiple sequencing lanes (with the same SeqType).\n- **Lane:** A number indicating which lane the library was sequenced on. Files\n  from the libraries sequenced on different lanes (and different SeqType) will\n  be concatenated after read clipping and merging.\n- **Colour Chemistry** A number indicating whether the Illumina sequencer the\n  library was sequenced on was a 2 (e.g. Next/NovaSeq) or 4 (Hi/MiSeq) colour\n  chemistry machine. This informs whether poly-G trimming (if turned on) should\n  be performed.\n- **SeqType:** A text string of either 'PE' or 'SE', specifying paired end (with\n  both an R1 [or forward] and R2 [or reverse]) and single end data (only R1\n  [forward], or BAM). This will affect lane merging if different per library.\n- **Organism:** A text string of the organism name of the sample or 'NA'. This\n  currently has no functionality and can be set to 'NA', but will affect\n  lane/library merging if different per library\n- **Strandedness:** A text string indicating whether the library type is\n  'single' or 'double'. This will affect lane/library merging if different per\n  library.\n- **UDG_Treatment:** A text string indicating whether the library was generated\n  with UDG treatment - either 'full', 'half' or 'none'. Will affect lane/library\n  merging if different per library.\n- **R1:** A text string of a file path pointing to a forward or R1 FASTQ file.\n  This can be used with the R2 column. File names **must be unique**, even if\n  they are in different directories.\n- **R2:** A text string of a file path pointing to a reverse or R2 FASTQ file,\n  or 'NA' when single end data. This can be used with the R1 column. File names\n  **must be unique**, even if they are in different directories.\n- **BAM:** A text string of a file path pointing to a BAM file, or 'NA'. Cannot\n  be specified at the same time as R1 or R2, both of which should be set to 'NA'\n\nFor example, the following TSV table:\n\n| Sample_Name | Library_ID | Lane | Colour_Chemistry | SeqType | Organism | Strandedness | UDG_Treatment | R1                                                             | R2                                                             | BAM |\n|-------------|------------|------|------------------|---------|----------|--------------|---------------|----------------------------------------------------------------|----------------------------------------------------------------|-----|\n| JK2782      | JK2782     | 7    | 4                | PE      | Mammoth  | double       | full          | data/JK2782_TGGCCGATCAACGA_L007_R1_001.fastq.gz.tengrand.fq.gz | data/JK2782_TGGCCGATCAACGA_L007_R2_001.fastq.gz.tengrand.fq.gz | NA  |\n| JK2782      | JK2782     | 8    | 4                | PE      | Mammoth  | double       | full          | data/JK2782_TGGCCGATCAACGA_L008_R1_001.fastq.gz.tengrand.fq.gz | data/JK2782_TGGCCGATCAACGA_L008_R2_001.fastq.gz.tengrand.fq.gz | NA  |\n| JK2802      | JK2802     | 7    | 4                | PE      | Mammoth  | double       | full          | data/JK2802_AGAATAACCTACCA_L007_R1_001.fastq.gz.tengrand.fq.gz | data/JK2802_AGAATAACCTACCA_L007_R2_001.fastq.gz.tengrand.fq.gz | NA  |\n| JK2802      | JK2802     | 8    | 4                | SE      | Mammoth  | double       | full          | data/JK2802_AGAATAACCTACCA_L008_R1_001.fastq.gz.tengrand.fq.gz | NA                                                             | NA  |\n\nwill have the following effects:\n\n- After AdapterRemoval, and prior to mapping, FASTQ files from lane 7 and lane 8\n  _with the same `SeqType`_ (and all other _metadata_ columns) will be\n  concatenated together for each **Library**.\n- After mapping, and prior BAM filtering, BAM files with different\n  `SeqType` (but with all other metadata columns the same) will be merged\n  together for each **Library**.\n- After duplicate removal, BAM files with different `Library_ID`s but with the same\n  `Sample_Name` and the same `UDG_Treatment` will be merged together.\n- If BAM trimming is turned on, all post-trimming BAMs (i.e. non-UDG and\n  half-UDG ) will be merged with UDG-treated (untreated) BAMs, if they have the\n  same `Sample_Name`.\n\nNote the following important points and limitations for setting up:\n\n- The TSV must use actual tabs (not spaces) between cells.\n- *File* names must be unique regardless of file path, due to risk of\n  over-writing (see:\n  [https://github.com/nextflow-io/nextflow/issues/470](https://github.com/nextflow-io/nextflow/issues/470)).\n  - If it is 'too late' and you already have duplicate file names, a workaround is\n    to concatenate the FASTQ files together and supply this to a nf-core/eager\n    run. The only downside is that you will not get independent FASTQC results\n    for each file.\n- Lane IDs must be unique for each sequencing of each library.\n  - If you have a library sequenced e.g. on Lane 8 of two HiSeq runs, you can\n    give a fake lane ID (e.g. 20) for one of the FASTQs, and the libraries will\n    still be processed correctly.\n  - This also applies to the SeqType column, i.e. with the example above, if one\n    run is PE and one run is SE, you need to give fake lane IDs to one of the\n    runs as well.\n- All _BAM_ files must be specified as `SE` under `SeqType`.\n  - You should provide a small decoy reference genome with pre-made indices, e.g.\n    the human mtDNA or phiX genome, for the mandatory parameter `--fasta` in\n    order to avoid long computational time for generating the index files of the\n    reference genome, even if you do not actually need a reference genome for any\n    downstream analyses.\n- nf-core/eager will only merge multiple _lanes_ of sequencing runs with the\n  same single-end or paired-end configuration\n- Accordingly nf-core/eager will not merge _lanes_ of FASTQs with BAM files\n  (unless you use `--run_convertbam`), as only FASTQ files are lane-merged\n  together.\n- Same libraries that are sequenced on different sequencing configurations (i.e\n  single- and paired-end data), will be merged after mapping and will _always_\n  be considered 'paired-end' during downstream processes\n  - **Important** running DeDup in this context is _not_ recommended, as PE and\n    SE data at the same position will _not_ be evaluated as duplicates.\n    Therefore not all duplicates will be removed.\n  - When you wish to run PE/SE data together `-dedupper markduplicates` is\n    therefore preferred.\n  - An error will be thrown if you try to merge both PE and SE and also supply\n    `--skip_merging`.\n  - If you truly want to mix SE data and PE data but using mate-pair info for PE\n    mapping, please run FASTQ preprocessing mapping manually and supply BAM\n    files for downstream processing by nf-core/eager\n  - If you _regularly_ want to run the situation above, please leave a feature\n    request on github.\n- DamageProfiler, NuclearContamination, MTtoNucRatio and PreSeq are performed on\n  each unique library separately after deduplication (but prior same-treated\n  library merging).\n- nf-core/eager functionality such as `--run_trim_bam` will be applied to only\n  non-UDG (UDG_Treatment: none) or half-UDG (UDG_Treatment: half) libraries.\n- Qualimap is run on each sample, after merging of libraries (i.e. your values\n  will reflect the values of all libraries combined - after being damage trimmed\n  etc.).\n- Genotyping will be typically performed on each `sample` independently, as\n  normally all libraries will have been merged together. However, if you have a\n  mixture of single-stranded and double-stranded libraries, you will normally\n  need to genotype separately. In this case you **must** give each the SS and DS\n  libraries _distinct_ `Sample_IDs`; otherwise you will receive a `file\n  collision` error in steps such as `sexdeterrmine`, and then you will need to\n  merge these yourself. We will consider changing this behaviour in the future\n  if there is enough interest."
```